py/simple_reporter.py

```diff
@@ -135,54 +140,75 @@ def match(file_names, config, quantisation, source, dest_dir):
 
     #do each trace in this file
     tiles = {}
-    for uuid, points in traces.iteritems():
-      #skip short traces
-      if len(points) < 2:
-        continue
-
+    for uuid, all_points in traces.iteritems():
       #sort the points by time in case threads were competing on the file
-      points.sort(key=lambda v:v['time'])
+      all_points.sort(key=lambda v:v['time'])
       report_levels = set([0, 1])
       transition_levels = set([0, 1])
       threshold_sec = 15
-      trace = {'uuid': uuid, 'trace': points}
-      
-      #get the matches for it
-      try:
-        match_str = segment_matcher.Match(json.dumps(trace, separators=(',', ':')))
-        match = json.loads(match_str)
-        report = reporter_service.report(match, trace, threshold_sec, report_levels, transition_levels)
-      except:
-        logger.error('Failed to report trace from %s: %s' % (file_name, json.dumps(trace, separators=(',', ':'))))
-        continue
-      
-      #weed out the usable segments and then send them off to the time tiles
-      segments = [ r for r in report['datastore']['reports'] if r['t0'] > 0 and r['t1'] > 0 and r['t1'] - r['t0'] > .5 and r['length'] > 0 and r['queue_length'] >= 0 ]
-      for r in segments:
-        duration = int(round(r['t1'] - r['t0']))
-        start = int(math.floor(r['t0']))
-        end = int(math.ceil(r['t1']))
-        min_bucket = int(start / quantisation)
-        max_bucket = int(end / quantisation)
-        for b in range(min_bucket, max_bucket + 1):
-          tile_level = str(get_tile_level(r['id']))
-          tile_index = str(get_tile_index(r['id']))
-          tile_file_name = dest_dir + os.sep + str(b * quantisation) + '_' + str((b + 1) * quantisation - 1) + os.sep + tile_level + os.sep + tile_index
-          s = [
-            str(r['id']), str(r.get('next_id', INVALID_SEGMENT_ID)), str(duration), '1',
-            str(r['length']), str(r['queue_length']), str(start), str(end), source, 'AUTO'
-          ]
-          tiles.setdefault(tile_file_name, []).append(','.join(s) + os.linesep)
+
+      #find the points where inactivity occurs, these are the window boundaries
+      windows = []
+      for i, point in enumerate(all_points):
+        if i == 0 or point['time'] - points[i - 1]['time'] > inactivity:
```

Comment:
here we find where there was inactivity in the trace and mark the index. we then only match windows where there arent large gaps of inactivity